graylog2-server/src/main/java/org/graylog2/entitygroups/rest/EntityGroupResource.java

```diff
@@ -0,0 +1,207 @@
+import org.graylog2.database.PaginatedList;
+import org.graylog2.plugin.rest.PluginRestResource;
+import org.graylog2.rest.models.SortOrder;
+import org.graylog2.shared.rest.resources.RestResource;
+
+import jakarta.inject.Inject;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.DefaultValue;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.graylog2.shared.utilities.StringUtils;
+
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_CREATE;
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_DELETE;
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_EDIT;
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_READ;
+import static org.graylog2.shared.utilities.ExceptionUtils.getRootCauseMessage;
+
+@Api(value = "EntityGroups", description = "Manage Entity Groups")
+@Path("/entity_groups")
+@RequiresAuthentication
+@Produces(MediaType.APPLICATION_JSON)
+public class EntityGroupResource extends RestResource implements PluginRestResource {
+
+    private final EntityGroupService entityGroupService;
+
+    @Inject
+    public EntityGroupResource(EntityGroupService entityGroupService) {
+        this.entityGroupService = entityGroupService;
+    }
+
+    @GET
+    @ApiOperation(value = "Get a list of entity groups")
+    @RequiresPermissions(ENTITY_GROUP_READ)
+    public PaginatedList<EntityGroup> listGroups(@ApiParam(name = "page") @QueryParam("page") @DefaultValue("1") int page,
+                                                 @ApiParam(name = "per_page") @QueryParam("per_page") @DefaultValue("15") int perPage,
+                                                 @ApiParam(name = "query") @QueryParam("query") @DefaultValue("") String query,
+                                                 @ApiParam(name = "sort",
+                                                           value = "The field to sort the result on",
+                                                           allowableValues = "name")
+                                                 @QueryParam("sort") String sort,
+                                                 @ApiParam(name = "direction", value = "The sort direction", allowableValues = "asc,desc")
+                                                 @DefaultValue("asc") @QueryParam("direction") SortOrder order) {
+        return entityGroupService.findPaginated(query, page, perPage, order, sort, null);
+    }
+
+    @GET
+    @Path("/{id}")
+    @ApiOperation(value = "Get a single entity group")
+    @ApiResponses(@ApiResponse(code = 404, message = "No such entity group"))
+    @RequiresPermissions(ENTITY_GROUP_READ)
```

Comment:
You shouldn't use the annotation if you check the individual permission in the code as that might confuse future readers.
Another PR will add marker annotations to clearly spell out the intent, but for now, let's use either an annotation or the code check, but not both at the same time (also reduces the risk of modifying one but not the other).

Reply:
Ok definitely makes sense, have removed the extraneous annotations in this class.